### PR TITLE
[WIP] Add particle thermal boundary condition

### PR DESCRIPTION
**Pull request by RemiLehe**
TODO:
- [ ] Prevent laser particles from calling the boundary condition routines (they should just be absorbed)
- [ ] Test with a laser pulse

### CODE REVIEW

File: Source/Particles/ParticleBoundaries_K.H [hunk 1]
```diff
@@ -64,6 +64,76 @@ namespace ApplyParticleBoundaries {
         }
     }
 
+
+namespace {
```
Comment (RemiLehe) [inline Source/Particles/ParticleBoundaries_K.H:68]: This function should be in a separate `.H` file.

**Comment by archermarx** [comment]
Neat! My understanding is this is effectively a diffuse thermal reflection BC, so particles that hit the boundary have their velocities resampled according to a provided thermal velocity?

**Comment by RemiLehe** [comment]
@archermarx Yes, that's correct.
It is supposed to model what happens if the simulation box was embedded in an infinite plasma with a finite temperature.
The PR also includes some changes for the field boundary conditions, which I will later extract in a separate PR to clean things up.
